ros_ws/src/teleoperation/src/keyboard_controller.cpp

```diff
@@ -181,12 +209,13 @@ void KeyboardController::publishDriveParameters()
     this->m_drive_parameters_publisher.publish(drive_parameters);
 }
 
-void KeyboardController::unlockMotorCallback(const std_msgs::Bool::ConstPtr& unlock_motor_message)
+void KeyboardController::driveModeCallback(const std_msgs::Int32::ConstPtr& drive_mode_message)
 {
-    if (this->m_car_unlocked != unlock_motor_message->data)
+    auto mode = (DriveMode)drive_mode_message->data;
+    if (this->m_drive_mode != mode)
     {
-        this->m_car_unlocked = unlock_motor_message->data;
-        this->updateWindowColor();
+        this->m_drive_mode = mode;
```

Comment:
Here you could also add an assertion to make sure the given data is indeed a valid DriveMode value.